mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebChromeClient.java

```diff
@@ -55,6 +65,74 @@ public boolean onConsoleMessage(final ConsoleMessage message) {
         return super.onConsoleMessage(message);
     }
 
+    public void openFileChooser(ValueCallback<Uri> uploadFile, String acceptType, String capture) {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+
+        Log.d("Opening file chooser");
+
+        try {
+            mIitc.startActivityForResult(intent, new FileRequestHandlerBelowLollipop(uploadFile));
+        } catch (final ActivityNotFoundException e) {
+            Toast.makeText(mIitc, "No activity to select a file found." +
+                    "Please install a file browser of your choice!", Toast.LENGTH_LONG).show();
+        }
+    }
+
+    private class FileRequestHandlerBelowLollipop implements IITC_Mobile.ResponseHandler {
+        FileRequestHandlerBelowLollipop(ValueCallback<Uri> filePathCallback) {
+            this.filePathCallback = filePathCallback;
+        }
+
+        ValueCallback<Uri> filePathCallback;
+
+        @Override
+        public void onActivityResult(final int resultCode, final Intent data) {
+            mIitc.deleteResponseHandler(this); // to enable garbage collection
+
+            Uri uri = data.getData();
+            if (uri != null) {
+                this.filePathCallback.onReceiveValue(uri);
+            }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    @Override
+    public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, FileChooserParams fileChooserParams) {
+        if (fileChooserParams.isCaptureEnabled()) {
+            Toast.makeText(mIitc, "Live Capture is not supported currently", Toast.LENGTH_LONG).show();
```

Comment:
Does it need some special support?

Reply:
I think capture would require using of camera or microphone to create some temp files. Are these functions needed by IITC?

Reply:
I suppose implementation is up on system, and we should not care.
I am if `capture` could be really useful for some plugin, but if it does not require any special support - why forbid it?